### ISSUE TITLE
switching language loses product

### ISSUE THREAD

**Issue by saliozzia** [loadedcommerce/loaded7#566]
when switching the language setting in the catalog, 
1. while on a product page
2. when the product has different values for permalink - due to localized permalink values

the page reloads and product is not found.

- solution - attempt to locate the page based on the permalink text alone if you cannot locate it for the current language.

**Comment by geraldbullard** [comment]
In m testing on Latest Pro (7.003.3.3) with different values for permalink:

En: burberry-shirt
Es: burberry-shirt-es

The product page is not lost and refreshes to the page. http://grabilla.com/05506-da26befd-dd45-43d3-abd5-7cdc0dcea76b.png

Though there is still some logic flaws with the Image and Stock etc... They are not the original issue of losing the product altogether.

**Comment by wa4u** [comment]
Below my findings are on 7.003.4.1-RC1

If Permalink are like you said;
En: burberry-shirt
Es: burberry-shirt-es

It fail to reload the product for selected language.

If Permalink are like
En: burberry-shirt
Es: burberry-shirt

It works.

**Comment by wa4u** [comment]
I come out with a solution for the issue.

File: includes/classes/product.php

line No: 80

Add below code:

```php
	$QproductID = $lC_Database->query('select products_id as id from :table_products_description where products_keyword=:products_keyword');
        $QproductID->bindTable(':table_products_description', TABLE_PRODUCTS_DESCRIPTION);
        $QproductID->bindValue(':products_keyword', $id);
        $QproductID->execute();
```

Update query below: 
Replace  
pd.products_keyword = :products_keyword 
with
pd.products_id = :products_id

```php
$Qproduct = $lC_Database->query('select p.products_id as id, p.parent_id, p.products_quantity as quantity, p.products_price as price, p.products_model as model, p.products_tax_class_id as tax_class_id, p.products_weight as weight, p.products_weight_class as weight_class_id, p.products_date_added as date_added, p.manufacturers_id, p.has_children, pd.products_name as name, pd.products_blurb as blurb, pd.products_description as description, pd.products_keyword as keyword, pd.products_tags as tags, pd.products_url as url, p.products_sort_order as sort_order from :table_products p, :table_products_description pd where pd.products_keyword = :products_keyword and pd.language_id = :language_id and pd.products_id = p.products_id and p.products_status = :products_status');
```

Replace
$Qproduct->bindValue(':products_keyword', $id);
with
$Qproduct->bindValue(':products_id', $QproductID->valueInt('id'));

**Comment by wa4u** [comment]
Merged into 7.003.4.2
